component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -284,6 +294,30 @@ private boolean validateJWTInDataBase(String jti, long currentTimeInMillis,
         return true;
     }
 
+    /**
+     * This method is to get JTI from the DB.
+     * For the migration purposes (preserve existing behaviour),
+     * We are searching for the current tenant and default tenant.
+     *
+     * @param jti      JTI.
+     * @param tenantId Tenant id.
+     * @return JWT entry if exists.
+     * @throws OAuthClientAuthnException OAuthClientAuthnException.
+     */
+    private JWTEntry getJTIfromDB(String jti, final int tenantId) throws OAuthClientAuthnException {
+
+        List<JWTEntry> jwtEntries = jwtStorageManager.getJwtsFromDB(jti, tenantId);
+
+        if (jwtEntries.isEmpty()) {
+            return null;
+        }
+        // If there is only one entry return it.
+        if (jwtEntries.size() == 1) {
```

Comment:
Do we need to check whether tenant id of the jwt entries is equal to `tenantId`.

Reply:
No need. The SQL query will fetch the required tenant ID or -1. So if there is one entry then it will probably require tenant id or -1.